app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -73,6 +79,24 @@ export type MetamaskNotificationsControllerState = {
    * List of read metamask notifications
    */
   metamaskNotificationsReadList: string[];
+  /**
+   * Flag that indicates that the creating notifications is in progress
+   */
+  isUpdatingMetamaskNotifications: boolean;
+  /**
+   * Flag that indicates that the fetching notifications is in progress
+   * This is used to show a loading spinner in the UI
+   * when fetching notifications
+   */
+  isFetchingMetamaskNotifications: boolean;
```

Comment:
Future PR - lets refactor and clean up data-fetching and state management.

If we want loading markers on controllers, I kind of want the same structure for all our controllers.

Reply:
Will create a ticket.

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-654